src/source/Sdp/Deserialize.c

```diff
@@ -74,74 +96,80 @@ STATUS deserializeSessionDescription(PSessionDescription pSessionDescription, PC
 {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
-    PCHAR curr, tail, next;
-    UINT32 lineLen;
-    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
+    SdpResult_t sdpResult = SDP_RESULT_OK;
+    SdpDeserializerContext_t ctx;
+    CHAR* pValue;
+    SIZE_T valueLength;
+    UINT8 type;
+    SIZE_T minLength;
 
-    curr = sdpBytes;
-    tail = sdpBytes + STRLEN(sdpBytes);
+    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
 
-    while ((next = STRNCHR(curr, tail - curr, '\n')) != NULL) {
-        lineLen = (UINT32) (next - curr);
+    sdpResult = SdpDeserializer_Init(&ctx, sdpBytes, STRLEN(sdpBytes));
+    CHK(sdpResult == SDP_RESULT_OK, sdpResult);
 
-        if (lineLen > 0 && curr[lineLen - 1] == '\r') {
-            lineLen--;
-        }
+    for (; sdpResult == SDP_RESULT_OK;) {
+        sdpResult = SdpDeserializer_GetNext(&ctx, &type, (const CHAR**) &pValue, (size_t*) &valueLength);
 
-        if (0 == STRNCMP(curr, SDP_MEDIA_NAME_MARKER, (ARRAY_SIZE(SDP_MEDIA_NAME_MARKER) - 1))) {
-            CHK_STATUS(parseMediaName(pSessionDescription, curr, lineLen));
+        if (sdpResult == SDP_RESULT_OK) {
```

Comment:
I think this if can be removed, as this is expected behavior and flow will anyway continue.

Reply:
I put it here because we break in all other conditions. Otherwise it would be like below, which is basically same. I prefer to put SDP_RESULT_OK in first condition because it's the highest possible return value from SDP library. Then we can speed this up with this sequence.
```
if (sdpResult == SDP_RESULT_MESSAGE_END) {
    /* Reset return value when done. */
    retStatus = STATUS_SUCCESS;
    break;
} else if (sdpResult != SDP_RESULT_OK) {
    retStatus = sdpResult;
    break;
} else {
    /* Do nothing. */
}
```